backends/ent/retrieve.go

```diff
@@ -28,50 +34,109 @@ func (backend *Backend) Retrieve(id string, _ *storage.RetrieveOptions) (*sbom.D
 		backend.Options = NewBackendOptions()
 	}
 
-	entDoc, err := backend.client.Document.Query().
-		Where(document.HasMetadataWith(metadata.IDEQ(id))).
-		WithMetadata().
-		WithNodeList().
-		Only(backend.ctx)
+	documents, err := backend.GetDocumentsByID(id)
 	if err != nil {
-		return nil, fmt.Errorf("eager loading document edges: %w", err)
+		return nil, fmt.Errorf("querying documents: %w", err)
 	}
 
-	entDoc.Edges.Metadata, err = entDoc.QueryMetadata().
-		WithAuthors().
-		WithDocumentTypes().
-		WithTools().
-		Only(backend.ctx)
-	if err != nil {
-		return nil, fmt.Errorf("eager loading metadata edges: %w", err)
+	if len(documents) > 1 {
+		return nil, fmt.Errorf("%w %s", errMultipleDocuments, id)
+	}
+
+	return documents[0], nil
+}
+
+func (backend *Backend) GetDocumentsByID(ids ...string) ([]*sbom.Document, error) {
+	documents := []*sbom.Document{}
+
+	query := backend.client.Document.Query().
+		WithMetadata().
+		WithNodeList()
+
+	if len(ids) > 0 {
+		query.Where(document.IDIn(ids...))
 	}
 
-	entDoc.Edges.NodeList, err = entDoc.QueryNodeList().
-		WithNodes().
-		Only(backend.ctx)
+	entDocs, err := query.All(backend.ctx)
 	if err != nil {
-		return nil, fmt.Errorf("eager loading node list edges: %w", err)
+		return nil, fmt.Errorf("querying documents table: %w", err)
+	}
+
+	if len(entDocs) == 0 {
+		return nil, fmt.Errorf("%w %s", errNoDocuments, ids)
```

Comment:
Why would a zero length result be an error? Shouldn't the return values be an empty `[]*sbom.Document`  and nil?

Reply:
The intent here was that code that calls this method would be expecting to get query results whether IDs are passed in or not. Returning an error was a way of extracting common code from the `Retrieve` method and have error context to return. So by extension, if `Retrieve` doesn't get a result for the ID explicitly provided, should it return `nil, nil` or just `&sbom.Document{}, nil` instead of its current behavior? 

I guess this error checking can be done instead by the calling context (e.g. the `Retrieve` method or `bomctl`) if that makes more sense to you; want me to make the change?